### PR TITLE
Add C++ template

### PR DESCRIPTION
**Pull request by vbe0201**
This pull request adds a C++ template.

Source files are pretty much identical to the ones from the C template.

Build systems:

* [GNU Make](https://www.gnu.org/software/make/)

* [CMake](https://cmake.org/)

* [Bazel](https://bazel.build/)

Users can cherry-pick whatever they like and this might also be a solid start if they decide to familiarize themselves with one of them.

### CODE REVIEW

File: cpp/BUILD [hunk 1]
```diff
@@ -0,0 +1,11 @@
+licenses(["notice"])
```
Comment (ef4203) [inline cpp/BUILD]: Why did you include this file?

File: cpp/WORKSPACE [hunk 2]
```diff
@@ -0,0 +1 @@
+workspace(name = "example")
```
Comment (ef4203) [inline cpp/WORKSPACE]: Why did you include this file?

File: cpp/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,11 @@
+cmake_minimum_required(VERSION 3.0)
+
+project(example CXX)
```
Comment (ef4203) [inline cpp/CMakeLists.txt]: Is there are way to circumvent using "example"? If this placeholder project name was included in multiple locations, one using the boilerplate would change it in multiple locations

File: cpp/src/main.cpp [hunk 4]
```diff
@@ -0,0 +1,8 @@
+#include <iostream>
+
+#include "a.h"
+
+int main(int argc, char **argv) {
+    std::cout << "Hello World!\n";
```
Comment (ef4203) [inline cpp/src/main.cpp]: This is grammatically incorrect, it should be "Hello, World!"

File: cpp/CMakeLists.txt (hunk 3 above)
Comment (vbe0201) [reply cpp/CMakeLists.txt]: There isn't. As stated [here](https://cmake.org/cmake/help/latest/command/project.html), every top-level `CMakeLists.txt` file must contain a direct call to `project()`.

File: cpp/BUILD (hunk 1 above)
Comment (vbe0201) [reply cpp/BUILD]: The `BUILD` and `WORKSPACE` file both correspond to [Bazel](https://bazel.build/). It is easy to use, feature-rich, has great documentation and is pretty widespread for modern C++. Companies like Google do excessively use it for good reason.

File: cpp/src/main.cpp (hunk 4 above)
Comment (vbe0201) [reply cpp/src/main.cpp]: Fixed.

**Comment by ef4203** [comment]
Thank you Valentin for your contribution, the usage of CMake is very appreciated and will be used in the the C template as well. Though the usage of Bazel is rejected, for reasons of keeping the boilerplate [MVP](https://en.wikipedia.org/wiki/Minimum_viable_product).

**Pull request merged**